### PR TITLE
[User] Answers-questions

### PR DESCRIPTION
**Pull request by QuangVinh93**

### CODE REVIEW

File: app/views/categories/_left.html.erb [hunk 1]
```diff
@@ -17,9 +16,9 @@
       </div>
     </li>
     <li>
-      <div class="btn btn-default col-md-3 col-md-offset-9">
-        <%= link_to t "views.categories.start" %>
-      </div>
+        <%= link_to t("views.categories.start"),
```
Comment (HUSTToanLH) [inline app/views/categories/_left.html.erb]: ident

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -1,15 +1,42 @@
...
+    end
+  end
+
+  def get_category_name category_id
+    category = Category.find_by id: category_id
+    category.name
+  end
+
+  def get_word_answers word_id
+    WordAnswer.send(:by_word_id, word_id)
```
Comment (HUSTToanLH) [inline app/helpers/application_helper.rb]: remove()

File: app/controllers/lessons_controller.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
+class LessonsController < ApplicationController
+  before_action :current_lesson, only: [:update, :show]
+
+  def update
+    if @lesson.result.nil?
+      @lesson.update_attributes result: 0
+      @lesson.update_attributes lesson_params
+      @lesson.lesson_words.each do |lesson_word|
+        result = check_result(lesson_word.word_id, lesson_word.word_answer_id)
```
Comment (HUSTToanLH) [inline app/controllers/lessons_controller.rb]: remove()

File: app/controllers/lessons_controller.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    else
+      @lesson.lesson_words.each do |lesson_word|
+        @words.push Word.find_by id: lesson_word.word_id
+      end
+    end
+  end
+
+  private
+  def lesson_params
+    params.require(:lesson).permit(:user_id, :category_id, :result,
```
Comment (HUSTToanLH) [inline app/controllers/lessons_controller.rb]: remove()

File: app/views/lessons/_answer_sheet.html.erb [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    <% get_word_answers(@words.at(lesson_word.index).id).
+      each do |answer| %>
+      <li>
+        <%= lesson_word.radio_button :word_answer_id, answer.id %>
+        <%= answer.content %>
+      </li>
+    <% end %>
+  </ul>
+  <% end %>
+  <%= f.submit "Submit" %>
```
Comment (hellovietnam93) [inline app/views/lessons/_answer_sheet.html.erb]: I18n

File: app/controllers/lessons_controller.rb [hunk 6]
```diff
@@ -0,0 +1,50 @@
+class LessonsController < ApplicationController
+  before_action :load_lesson, only: [:update, :show, :create]
+
+  def create
+    lesson = current_user.lessons.build category_id: params[:category_id]
+    Settings.QUESTION_PER_LESSON.times{
+      lesson.lesson_words.build
+    }
+    lesson.save
+    redirect_to lesson_path lesson.id
```
Comment (hellovietnam93) [inline app/controllers/lessons_controller.rb]: redirect_to lesson

File: app/models/word_answer.rb [hunk 7]
```diff
@@ -1,5 +1,7 @@
 class WordAnswer < ActiveRecord::Base
   belongs_to :word
-  
+
   has_many :lesson_words
+
+  scope :by_word_id, ->word_id{where word_id: word_id}
```
Comment (hellovietnam93) [inline app/models/word_answer.rb]: cái này dùng ở chỗ nào ?

File: app/views/lessons/_answer_sheet.html.erb [hunk 8]
```diff
@@ -0,0 +1,22 @@
+<h2 class="text-center">
+  <strong><%= t "views.lessons.questions" %></strong>
+  <hr>
+</h2>
+<%= form_for @lesson do |f| %>
+  <%= f.fields_for :lesson_words do |lesson_word| %>
+  <ul class="list-unstyled">
```
Comment (hellovietnam93) [inline app/views/lessons/_answer_sheet.html.erb]: ident

File: app/views/lessons/_result.html.erb [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+  </h3>
+</div>
+<div class="col-md-12">
+  <div class="col-md-2 col-md-2">
+    <ul class="list-unstyled">
+      <% @lesson.lesson_words.each do |lesson_word| %>
+        <% if lesson_word.word_answer.nil? %>
+          <li><%= t "views.lessons.wrong_answer" %></li>
+        <% else %>
+          <% if lesson_word.word_answer.correct %>
```
Comment (hellovietnam93) [inline app/views/lessons/_result.html.erb]: lesson_word.word_answer.correct ? t("views.lessons.true_answer") : t("views.lessons.wrong_answer")

File: app/views/lessons/show.html.erb [hunk 10]
```diff
@@ -0,0 +1,8 @@
+<%= provide :title, t("views.lessons.lessons") %>
+<div class="col-lg-8 col-lg-offset-2">
+  <% if @lesson.result.nil? %>
```
Comment (hellovietnam93) [inline app/views/lessons/show.html.erb]: <%= render @lesson.result.nil? ? "answer_sheet" : "result" %>

File: app/views/lessons/_result.html.erb [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+<div class="col-md-12">
+  <div class="col-md-2 col-md-2">
+    <ul class="list-unstyled">
+      <% @lesson.lesson_words.each do |lesson_word| %>
+        <% if lesson_word.word_answer.nil? %>
+          <li><%= t "views.lessons.wrong_answer" %></li>
+        <% else %>
+            <li>
+              <%= lesson_word.word_answer.correct ?
+              t("views.lessons.true_answer") :
```
Comment (hellovietnam93) [inline app/views/lessons/_result.html.erb]: ident

File: app/views/lessons/_result.html.erb [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+<div class="col-md-12">
+  <div class="col-md-2 col-md-2">
+    <ul class="list-unstyled">
+      <% @lesson.lesson_words.each do |lesson_word| %>
+        <% if lesson_word.word_answer.nil? %>
+          <li><%= t "views.lessons.wrong_answer" %></li>
+        <% else %>
+          <li>
+            <%= lesson_word.word_answer.correct ?
+            t("views.lessons.true_answer") :
```
Comment (hellovietnam93) [inline app/views/lessons/_result.html.erb]: ident

**Pull request merged**